### PR TITLE
Remove partition tables use cache instead

### PR DESCRIPTION
**Pull request by aldavidson**
The partitions on the postcode_api_address table were not doing enough to improve performance, and were causing significant complications in the AddressBase import - for instance, locking tables for a non-trivial amount of time. 

This branch gets rid of the partitions, and introduces disk-based local file caching using a standard Django backend. This means that the first hit to any given URL on any given server instance will be as slow as it was, but the output will then be cached on disk for up to 5 days. 

When the full import job has run through, it will write a new CacheVersion record with the filename of the latest import file. The combination of last_addressbase_file and timestamp is used as a 'version' for the cache, hence writing a new CacheVersion effectively clears the cache. Size of the cache is limited to 1m entries (we currently serve around 5k unique URLs per day, so this is plenty for now)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ RUN curl --silent --show-error --retry 5 https://bootstrap.pypa.io/get-pip.py |
 # APP_HOME, if you change this variable make sure you update the files in docker/ too
 ENV APP_HOME /srv/postcodeinfo
 
-RUN mkdir -p /var/log/gunicorn /var/log/nginx/postcodeinfo
+RUN mkdir -p /var/log/gunicorn /var/log/nginx/postcodeinfo
```
Comment (mattpep) [inline Dockerfile]: a stray trailing space has made its way in

File: postcodeinfo/apps/postcode_api/caches/view_cache.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+# The CACHES setting KEY_FUNCTION will specify this as the 
+# function to call whenever it's constructing a cache key
+# The dynamic CacheVersion call means that we can easily
+# invalidate the whole cache by incrementing the version in
+# the DB, and it will be picked up straight away by all the
+# servers in the auto-scaling group (which is more tricky to
+# achieve with shell commands)
+def make_key(key, key_prefix, version):
+    return ':'.join([key_prefix, CacheVersion.objects.latest_as_string(), key])
```
Comment (mattpep) [inline postcodeinfo/apps/postcode_api/caches/view_cache.py]: It is strongly preferred for text files to be terminated with a newline character.

E.g. if a subsequent line is added then without this newline, 2 lines will appear in the diff output even though only 1 has a semantic change.

File: postcodeinfo/apps/postcode_api/management/commands/write_new_cache_version.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+class Command(BaseCommand):
+    args = 'latest_filename'
+
+    def handle(self, *args, **options):
+        latest_filename = None
+        if len(args) == 1:
+            latest_filename = args[0]
+
+        cv = CacheVersion(last_addressbase_file=latest_filename)
+        cv.save()
```
Comment (mattpep) [inline postcodeinfo/apps/postcode_api/management/commands/write_new_cache_version.py]: Needs a newline at the end.

File: scripts/download_and_import_all.sh [hunk 4]
```diff
@@ -11,3 +11,11 @@ echo "Local Authorities"
...
 echo "Postcode / GSS Codes"
 ./manage.py download_and_import_postcode_gss_codes
+
+echo "*******************************************"
+echo "writing new cache version"
+unset -v latest
+for file in "/tmp/addressbase_basic"/*; do
+  [[ $file -nt $latest ]] && latest=$file
+done
+./manage.py write_new_cache_version ${latest}
```
Comment (mattpep) [inline scripts/download_and_import_all.sh]: Needs a newline at the end.

File: postcodeinfo/apps/postcode_api/management/commands/write_new_cache_version.py (hunk 3 above)
Comment (aldavidson) [inline postcodeinfo/apps/postcode_api/management/commands/write_new_cache_version.py]: fixed

File: postcodeinfo/apps/postcode_api/caches/view_cache.py (hunk 2 above)
Comment (aldavidson) [inline postcodeinfo/apps/postcode_api/caches/view_cache.py]: fixed

**Comment by mattpep** [comment]
LGTM. Ready for a squish, then I'll merge

**Comment by aldavidson** [comment]
squished!

**Pull request merged**